straxen/plugins/afterpulse_processing.py

```diff
@@ -0,0 +1,345 @@
+                                 help='Right boundary of sample range for LED pulse integration',
+                                ),
+                    strax.Option('baseline_samples',
+                                 default=40,
+                                 help='Number of samples to use at start of WF to determine the baseline',
+                                ),
+                    strax.Option('hit_min_amplitude',
+                                 track=True,
+                                 default=('hit_thresholds_tpc', 'ONLINE', True),
+                                 help='Minimum hit amplitude in ADC counts above baseline. '
+                                      'Specify as a tuple of length n_tpc_pmts, or a number,'
+                                      'or a string like "pmt_commissioning_initial" which means calling'
+                                      'hitfinder_thresholds.py'
+                                      'or a tuple like (correction=str, version=str, nT=boolean),'
+                                      'which means we are using cmt.',
+                                ),
+                    strax.Option('hit_min_height_over_noise',
+                                 default=4,
+                                 help='Minimum hit amplitude in numbers of baseline_rms above baseline.'
+                                      'Actual threshold used is max(hit_min_amplitude, hit_min_height_over_noise * baseline_rms).',
+                                ),
+                    strax.Option('hit_left_extension',
+                                 default=2,
+                                 help='Extend hits by this many samples left',
+                                ),
+                    strax.Option('hit_right_extension',
+                                 default=20,
+                                 help='Extend hits by this many samples right',
+                                ),
+                   )
+class LEDAfterpulseProcessing(strax.Plugin):
+    
+    __version__ = '0.4.0'
+    depends_on = 'raw_records'
+    data_kind = 'afterpulses'
+    provides = 'afterpulses'
+    compressor = 'zstd'
+    parallel = 'process'
+    rechunk_on_save = True
+    
+    dtype = dtype_ap
+    
+    def setup(self):
+        
+        self.to_pe = straxen.get_correction_from_cmt(self.run_id, self.config['gain_model'])
+        
+        # Check config of `hit_min_amplitude` and define hit thresholds
+        # if cmt config
+        if is_cmt_option(self.config['hit_min_amplitude']):
+            self.hit_thresholds = straxen.get_correction_from_cmt(self.run_id,
+                self.config['hit_min_amplitude'])
+        # if hitfinder_thresholds config
+        elif isinstance(self.config['hit_min_amplitude'], str):
+            self.hit_thresholds = straxen.hit_min_amplitude(
+                self.config['hit_min_amplitude'])
+        else: # int or array
+            self.hit_thresholds = self.config['hit_min_amplitude']
```

Comment:
I agree that it looks quite annoying, but at this moment I have no idea how it can be done more elegantly... Any suggestion?

Reply:
Hi guys, 
I think the issue is because we want get the `hit_min_amplitude` from three different places, 1) CMT, 2) `hitfinder_thresholds.py` and 3) passing an int or array. Do we really need to support these three approaches? I would suggest we don't need to support `hitfinder_thresholds.py` anymore, we can use CMT for analysis and option 3) for testing.  Or did I missed something here @WenzDaniel

Reply:
Is there a decision on this yet? Then I can adjust accordingly